### PR TITLE
Add support to float

### PR DESCRIPTION
**Pull request by katcipis**
And some redesign

### CODE REVIEW

**Review by vitorarins — APPROVED** [review]
LGTM

File: waveparser_test.go [hunk 1]
```diff
@@ -100,3 +103,52 @@ func TestParseWAV(t *testing.T) {
...
+	assertNoError(t, err)
+
+	samples, err := wav.Int16LESamples()
+	assertNoError(t, err)
+
+	gotbuf := &bytes.Buffer{}
+	err = binary.Write(gotbuf, binary.LittleEndian, samples)
+	assertNoError(t, err)
+
+	expected, err := ioutil.ReadFile("testdata/audios/sint16le.raw")
```
Comment (i4ki) [inline waveparser_test.go:119]: if you say it's right then I believe =)

**Review by i4ki — APPROVED** [review]
LGTM

File: waveparser_test.go (hunk 1 above)
Comment (katcipis) [reply waveparser_test.go:119]: uaheuhuehuah

you can open the audio on audacity with "import raw audio"..no need for trust =P

**Pull request merged**